client/src/components/Visualization/charts/Geochart.js

```diff
@@ -1,76 +1,80 @@
 /**
  * Description: Geochart implements a Google geochart component.
  * Author: Kira Toal
- * Date: 2020/07/13
+ * Date: 2020/21/13
  */
-
-import React, { useEffect, useState } from 'react';
-import firebase, { app, database } from '../../../firebase/firebase';
-
-import { Chart } from 'react-google-charts';
-import { states } from './StateDataParser';
+import { Chart } from "react-google-charts";
+import React, { useState, useEffect } from 'react';
+import { STATES } from '../../../constants/geochart_constants';
+import { database } from '../../../firebase/firebase';
 
 const Geochart = () => {
+  const STATES_COLLECTION = 'dev_states';
   const [adTotal, setAdTotal] = useState(0);
   const WIDTH = '700';
   const HEIGHT = '400';
 
-  class GeochartAd {
-    constructor(id) {
-      /**
-       * @param {string} id
-       */
-      this.id = id;
-    }
-    toString() {
-      return this.id;
-    }
-  }
+  useEffect(() => {
+    async function fetchStateTotals() {
+      /* 
+      dataTable is read by the Chart component into Geochart format.
+      The states are colored according to the value in the second column of
+        the table ("Total Ad Spend (USD)").
+      The first column (state name) and third column (info on total state ad 
+        spend and top advertiser) are shown in the tooltip.
+      */ 
+      let dataTable = [["State", "Total Ad Spend (USD)", {type: 'string', role: 'tooltip'}]];
+      for (let state of STATES) {
+        
+        // Calculate state spend totals.
+        let stateTotal = 0;
+        let stateSpendData = await database.collection(STATES_COLLECTION)
```

Comment:
Why are the database.collection calls here being done sequentially with `await`s, especially since they're independent. Might be contributing to the long latency.

Reply:
Also general debugging advice: you should be able to use the "Networks" tab in chrome dev tools to track the duration of the server calls